### PR TITLE
update to target SDK 35 and dependencies

### PR DESCRIPTION
**Pull request by ashnohe**

### CODE REVIEW

**Comment by ashnohe** [comment]
@calren while updating dependencies, it looks like there was an API change for `DefaultPreloadManager.Status`. Can you confirm that `STAGE_LOADED_FOR_DURATION_MS` is the right one to use?

File: app/src/main/java/com/google/android/samples/socialite/ui/player/preloadmanager/PreloadManagerWrapper.kt [hunk 1]
```diff
@@ -160,7 +160,7 @@ private constructor(
     class PreloadStatusControl : TargetPreloadStatusControl<Int> {
         override fun getTargetPreloadStatus(rankingData: Int): DefaultPreloadManager.Status {
             // By default preload first 3 seconds of the video
-            return DefaultPreloadManager.Status(STAGE_LOADED_TO_POSITION_MS, 3000L)
+            return DefaultPreloadManager.Status(STAGE_LOADED_FOR_DURATION_MS, 3000L)
```
Comment (calren) [inline app/src/main/java/com/google/android/samples/socialite/ui/player/preloadmanager/PreloadManagerWrapper.kt:163]: @MayuriKhinvasara can you comment on whether this is the right update after updating to the latest media3 release?

**Review by mlykotom — APPROVED** [review]

File: app/src/main/java/com/google/android/samples/socialite/ui/player/preloadmanager/PreloadManagerWrapper.kt (hunk 1 above)
Comment (MayuriKhinvasara) [reply app/src/main/java/com/google/android/samples/socialite/ui/player/preloadmanager/PreloadManagerWrapper.kt:163]: There are a couple of changes to PreloadManager when updating to media3 1.5. We would have to test them before commits.

**Comment by MayuriKhinvasara** [comment]
> @calren while updating dependencies, it looks like there was an API change for `DefaultPreloadManager.Status`. Can you confirm that `STAGE_LOADED_FOR_DURATION_MS` is the right one to use?

Can we not include to update the media3 lib to 1.5 ? I can take a separate AI for that. We expect a couple of changes to functioning of PreloadManager in media3 upgrade to 1.5. Will have to test out all those use cases before blanket update

**Pull request merged**